geode-core/build.gradle

```diff
@@ -155,6 +155,8 @@ configurations {
 }
 
 dependencies {
+
+  //These bom dependencies are used to constrain the versions of the dependencies listed below
   compile(platform(project(':boms:geode-all-bom')))
   compileOnly(platform(project(':boms:geode-all-bom')))
```

Comment:
Is it worth checking the need of 'compile' and 'compileOnly' and perhaps get away from this?

Reply:
I will see if I can change compile to api.

It looks like we are using compileOnly with the bom because further down in the file we are pulling in spring-webmvc as a compileOnly dependency and we need to constrain the version.

I think we can pull the code that depends on springweb-mvc completely out of core, but I'd like to do that as a separate PR if thats ok.

Reply:
I changed compile to api. We can remove compileOnly as soon the code for geode-web is moved out of core.